doc/source/analysis.rst

```diff
@@ -1,5 +1,400 @@
+
+>>> g.neighborhood([0, 1], order=2)
+
+and to find the neighborhood size, there is :meth:`Graph.neighborhood_size`.
+
+Degrees
++++++++
+To compute the degree, in-degree, or out-degree of a node, use :meth:`Vertex.degree`, :meth:`Vertex.indegree`, and :meth:`Vertex.outdegree`:
+
+>>> deg = g.vs[0].degree()
+>>> deg = g.degree(0)
+
+To compute the max degree in a list of vertices, use :meth:`Graph.maxdegree`.
+
+:meth:`Graph.knn` computes the average degree of the neighbors.
+
+Adding and removing vertices and edges
+++++++++++++++++++++++++++++++++++++++
+
+To add nodes to a graph, use :meth:`Graph.add_vertex` and :meth:`Graph.add_vertices`:
+
+>>> g.add_vertex()
+>>> g.add_vertices(5)
+
+This changes the graph `g` in place. You can specify the name of the vertices if you wish.
+
+To remove nodes, use :meth:`Graph.delete_vertices`:
+
+>>> g.delete_vertices(None)  # remove all vertices
+>>> g.delete_vertices([1, 2])
+
+Again, you can specify the names or the actual :class:`Vertex` objects instead.
+
+To add edges, use :meth:`Graph.add_edge` and :meth:`Graph.add_edges`:
+
+>>> g.add_edge(0, 2)
+>>> g.add_edges([(0, 2), (1, 3)])
+
+To remove edges, use :meth:`Graph.delete_edges`:
+
+>>> g.delete_edges(None)   # remove all edges
+>>> g.delete_edges([0, 5]) # remove by edge ID
+
+You can also remove edges between source and target nodes.
+
+To contract vertices, use :meth:`Graph.contract_vertices`. Edges between contracted vertices will become loops.
+
+Graph operators
++++++++++++++++++
+It is possible to compute the union, intersection, difference, and other set operations (operators) between graphs.
+
+To compute the union of the graphs (nodes/edges in either are kept):
+
+>>> gu = ig.union([g, g2, g3])
+
+Similarly for the intersection (nodes/edges present in all are kept):
+
+>>> gu = ig.intersection([g, g2, g3])
+
+These two operations preserve attributes and can be performed with a few variations. The most important one is that vertices can be matched across the graphs by id (number) or by name.
+
+These and other operations are also available as methods of the :class:`Graph` class:
+
+>>> g.union(g2)
+>>> g.intersection(g2)
+>>> g.disjoint_union(g2)
+>>> g.difference(g2)
+>>> g.complementer()  # complement graph, same nodes but missing edges
+
+and even as numerical operators:
+
+>>> g |= g2
+>>> g_intersection = g and g2
+
+Topological sorting
++++++++++++++++++++
+To sort a graph topologically, use :meth:`Graph.topological_sorting`:
+
+>>> g = ig.Graph.Tree(10, 2, mode=ig.TREE_OUT)
+>>> g.topological_sorting()
+
+Graph traversing
++++++++++++++++++++++
+A common operation is traversing the graph. |igraph| currently exposes breath-first search (BFS) via :meth:`Graph.bfs` and :meth:`Graph.bfsiter`:
+
+>>> [vertices, layers, parents] = g.bfs()
+>>> it = g.bfsiter()  # Lazy version
+
+A depth-first search function is in the works.
```

Comment:
Note sure this comment is necessary ...

Can you create a "TODO" issue saying to update this line once DFS has been added? This is the kind of thing that tends to get forgotten. Then we end up with a tutorial that claims that some functionality is NOT available when in reality it is.

Reply:
Agreee - let's remove this or convert this into a TODO comment. Most IDEs can be configured to highlight and collect TODO items into a single panel so it won't get forgotten.

Reply:
That PR is done, so I'll just add DFS here and make this PR conditional to #315